lib/charms/operator_libs_linux/v1/systemd.py

```diff
@@ -36,18 +36,22 @@
 
 """
 
+import functools
 import logging
 import subprocess
 
 __all__ = [  # Don't export `_systemctl`. (It's not the intended way of using this lib.)
```

Comment:
While I am in the neighborhood, I noticed the having `__all__` declared after the stdlib module imports violates [PEP8](https://peps.python.org/pep-0008/#module-level-dunder-names). `__all__` should come before the module imports but after the module docstring. Easy fix!

Reply:
Fixed in commit https://github.com/canonical/operator-libs-linux/pull/106/commits/af8c1fb3b34fbfc19b74b20d4210dd45c7eb07db